ui/src/main/js/field/FieldMappingField.js

```diff
@@ -1,135 +1,126 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
 import TableDisplay from 'field/TableDisplay';
 import TextInput from 'field/input/TextInput';
-import LabeledField from 'field/LabeledField';
-
-class FieldMappingField extends Component {
-    constructor(props) {
-        super(props);
-
-        this.createNewRow = this.createNewRow.bind(this);
-        this.createColumns = this.createColumns.bind(this);
-        this.onEdit = this.onEdit.bind(this);
-        this.handleChange = this.handleChange.bind(this);
-        this.saveModalData = this.saveModalData.bind(this);
-        this.clearModal = this.clearModal.bind(this);
-        this.onDelete = this.onDelete.bind(this);
-
-        let currentId = 0;
+import LabeledField, { LabelFieldPropertyDefaults } from 'field/LabeledField';
+
+const FieldMappingField = ({
+    id,
+    description,
+    editMappingTitle,
+    errorName,
+    errorValue,
+    fieldMappingKey,
+    label,
+    labelClass,
+    leftSideMapping,
+    newMappingTitle,
+    onChange,
+    required,
+    rightSideMapping,
+    showDescriptionPlaceHolder,
+    storedMappings
+}) => {
+    const [fieldMappingRow, setFieldMappingRow] = useState({
+        rowId: -1,
+        fieldName: '',
+        fieldValue: ''
+    });
+    const [tableData, setTableData] = useState([]);
+    const [tableId, setTableId] = useState(0);
+    const [modalError, setModalError] = useState(null);
+    const [modalTitle, setModalTitle] = useState(newMappingTitle);
+    useEffect(() => {
+        let currenttableId = 0;
         let fieldMappings = [];
-        const { storedMappings } = props;
         if (storedMappings) {
             fieldMappings = storedMappings.map((mapping) => JSON.parse(mapping));
             fieldMappings.forEach((parsedMapping) => {
-                parsedMapping.id = currentId;
-                currentId += 1;
+                parsedMapping.tableId = currenttableId;
+                currenttableId += 1;
             });
         }
 
-        this.state = ({
-            fieldMappingRow: {
-                rowId: -1,
-                fieldName: '',
-                fieldValue: ''
-            },
-            tableData: fieldMappings,
-            id: currentId,
-            modalError: null,
-            modalTitle: props.newMappingTitle
-        });
-    }
+        setTableId(currenttableId);
+        setTableData(fieldMappings);
+    }, []);
 
-    handleChange({ target }) {
+    const handleChange = ({ target }) => {
```

Comment:
This function looks similar to FieldModelUtilities.handleTestChange. Maybe we can use that here as well?

Reply:
If so, we might want to change the shared function name

Reply:
It is similar but the contents of it aren't like a field model field.  It is a single table row that will eventually be added to a field model.  For now this will need to be the same but perhaps we can make a more abstract handler as we implement more UI changes.